cloudevents/work/agent/codec/manifest.go

```diff
@@ -29,12 +30,14 @@ const (
 
 // ManifestCodec is a codec to encode/decode a ManifestWork/cloudevent with ManifestBundle for an agent.
 type ManifestCodec struct {
-	restMapper meta.RESTMapper
+	sequenceGenerator *snowflake.Node
```

Comment:
this is only needed for status update, right？ I do not think a developer of consumer need to care about this.

Reply:
right, the agent generates the sequence id only when it creates the status update event.  

and creating the snowflake  generator when the agent init